ios/FluentUI/Core/SwiftUI+ViewPresentation.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Modifier that presents a SwiftUI View as an additional layer on top of the content View.
+struct PresentationModifier<PresentedView: View>: ViewModifier {
+    func body(content: Content) -> some View {
+        ZStack {
+            content
+
+            // Backdrop for the presented view which overlays the
+            // content where it's being presented from.
+            Rectangle()
+            .frame(maxWidth: .infinity,
+                   maxHeight: .infinity,
+                   alignment: .center)
+            .foregroundColor(isPresented ? backgroundColor : Color.clear)
+            .allowsHitTesting(isPresented && isBlocking)
+
+            presentedView
+        }
+        .frame(maxWidth: .infinity,
+               maxHeight: .infinity,
+               alignment: .center)
+    }
+
+    var hasDimmedBackground: Bool
+    var dimmedBackgroundColor: Color
+    var dimmedBackgroundColroOpacity: Double
+    var isBlocking: Bool
+    @Binding var isPresented: Bool
+    @ViewBuilder var presentedView: PresentedView
+
+    var backgroundColor: Color {
+        if hasDimmedBackground {
+            return dimmedBackgroundColor.opacity(dimmedBackgroundColroOpacity)
+        } else if isBlocking {
+            // A clear color won't allow the background
+            // view to block the underlying view
+            return Color.black.opacity(Double.ulpOfOne)
```

Comment:
Why ulpOfOne here? If that's the lowest possible double that's greater than 1.0, wouldn't 0.1 suffice/be more opaque?

Reply:
ulpOfOne is "The positive difference between 1.0 and the next greater representable number."
It's smaller than 0.1 and it's the best "clear" view that we can have if we don't use Color.clear.
The problem with Color.clear is that it seems to be "absence" of a View, which won't work to block the interaction.